paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/definitions/CardDefinition.kt

```diff
@@ -97,28 +100,60 @@ private object CardUiDefinitionFactory : UiDefinitionFactory.Simple {
                     )
                 )
             }
-            if (
-                isSaveForFutureUseValueChangeable(
-                    code = PaymentMethod.Type.Card.code,
-                    intent = metadata.stripeIntent,
-                    paymentMethodSaveConsentBehavior = metadata.paymentMethodSaveConsentBehavior,
-                    hasCustomerConfiguration = metadata.hasCustomerConfiguration,
-                )
-            ) {
+
+            val canChangeSaveForFutureUsage = saveForFutureUsageIsChangeable(metadata)
+
+            if (canChangeSaveForFutureUsage) {
                 add(SaveForFutureUseElement(arguments.saveForFutureUseInitialValue, arguments.merchantName))
             }
 
-            if (metadata.linkInlineConfiguration != null && arguments.linkConfigurationCoordinator != null) {
+            val usesLink = if (
+                metadata.linkInlineConfiguration != null && arguments.linkConfigurationCoordinator != null
+            ) {
                 add(
                     LinkFormElement(
                         configuration = metadata.linkInlineConfiguration,
                         linkConfigurationCoordinator = arguments.linkConfigurationCoordinator,
                         onLinkInlineSignupStateChanged = arguments.onLinkInlineSignupStateChanged,
                     )
                 )
+
+                true
+            } else {
+                false
+            }
+
+            if (metadata.hasIntentToSetup()) {
+                add(createCardMandateElement(metadata.merchantName, usesLink, canChangeSaveForFutureUsage))
             }
         }
     }
+
+    private fun createCardMandateElement(
+        merchantName: String,
+        usesLink: Boolean,
+        canChangeSaveForFutureUse: Boolean,
+    ): MandateTextElement {
+        return MandateTextElement(
+            identifier = IdentifierSpec.Generic("card_mandate"),
+            stringResId = PaymentSheetR.string.stripe_paymentsheet_card_mandate,
+            topPadding = when {
+                usesLink -> 0.dp
+                canChangeSaveForFutureUse -> 6.dp
+                else -> 8.dp
+            },
```

Comment:
Definitely don't think we should keep this. We should update the spacing so that this is handled automatically. I think it'll be easier to update though once we have the card mandate merged.